### PR TITLE
Issue/2605 bump db schema version

### LINKED ISSUE #2605: Bump the database schema version

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

* Bump DB schema version to allow patch release of the ISO3 and ISO4 DB schema.
* Add DB migration test (v7 -> v17)

closes #2605 

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation~~

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: tests/db/dumps/v17.sql [hunk 1]
```diff
@@ -0,0 +1,972 @@
...
+SET check_function_bodies = false;
+SET xmloption = content;
+SET client_min_messages = warning;
+SET row_security = off;
+
+--
+-- Name: plpgsql; Type: EXTENSION; Schema: -; Owner: -
+--
+
+CREATE EXTENSION IF NOT EXISTS plpgsql WITH SCHEMA pg_catalog;
```
Comment (wouterdb) [inline tests/db/dumps/v17.sql:23]: do we have this in our actual schema?

**Review by wouterdb — CHANGES_REQUESTED** [review]

File: tests/db/dumps/v17.sql (hunk 1 above)
Comment (arnaudsjs) [reply tests/db/dumps/v17.sql:23]: No. I think this is enabled by default.

**Review by wouterdb — APPROVED** [review]

**Review by bartv — APPROVED** [review]
It looks ok. however at least on the next bump we need to round the numbers because otherwise this gets way to complex. Like jump to 30 instead of 27

But leave it like this for now.

**Comment by arnaudsjs** [comment]
> It looks ok. however at least on the next bump we need to round the numbers because otherwise this gets way to complex. Like jump to 30 instead of 27
> 
> But leave it like this for now.

I agree.

**Pull request merged**